openquakeplatform/openquakeplatform/updatures/app.py

```diff
@@ -0,0 +1,931 @@
+                if isinstance(ref.model, types.FunctionType):
+                    continue
+                ref_md = models_descr[ref.model]
+                pdebug(2, "CC: REF_FIELD, REF.MODEL: %s, %s, %s" % (ref_field, ref.model, ref_md.natural))
+                ref_value = get_value(item, ref_field)
+                if not ref_value:
+                    continue
+
+                cm_out['fields_n'] += 1
+                ty = type(ref_value)
+
+                if ref_md.natural is not None:
+                    if ref.is_many:
+                        # one2many case
+                        refs_fk = ref_value
+                    else: # if ref.is_many:
+                        # any2one case:
+                        refs_fk = [ ref_value ]
+
+                    for fk in refs_fk:
+                        for fktem in dates_gr[ref.model]:
+                            if fk == ref_md.natural(fktem):
+                                break
+                        else:
+                            pdebug(1, "CC: natural")
+                            cm_out['incons'+('_is_many' if ref.is_many else '')] += 1
+
+                else: # if ref_md.natural
+                    pdebug(2, "CC: NOT NATURAL")
+
+                    # many2many case
+                    if ref.is_many:
+                        refs_fk = ref_value
+                    else:
+                        refs_fk = [ ref_value ]
+
+
+                    for fk in refs_fk:
+                        for fktem in dates_gr[ref.model]:
+                            if fk == fktem['pk']:
+                                break
+                        else:
+                            pdebug(1, "CC: not natural model: %s %s" % (model, item))
+                            cm_out['incons'+('_is_many' if ref.is_many else '')] += 1
+
+    return cm_out_gr
+
+
+def reference_get(dates_gr, model, pk):
+    md = models_descr[model]
+    for item in dates_gr[model]:
+        if md.natural:
+            if md.natural(item) == pk:
+                return item
+        else:
+            if item['pk'] == pk:
+                return item
+    else:
+        return None
+
+def item_key(model, item):
+    md = models_descr[model]
+    if md.natural:
+        return md.natural(item)
+    else:
+        return item['pk']
+
+def grouping_set(dates_gr, datesk_gr):
+    """
+    for composite data structure like vulnerability group records of different model
+    to the same object
+    """
+    group_heads = []
+
+    for model in models_order:
+        md = models_descr[model]
+        pdebug(2, "Mod: %s  Is grouped: %s" % (model, md.group))
+        if md.group is None or not dates_gr[model]:
+            pdebug(2, "No grouping or items for model %s" % model)
+            continue
+
+        if md.group not in group_heads:
+            group_heads.append(md.group)
+        dirref_groups = [(k,v) for k,v in md.refs.iteritems() if not isinstance(v.model, types.FunctionType) and v.model == md.group]
```

Comment:
The line looks longer than 79 characters, you can split it in two before the `if`